app/controllers/api/v1/keywords_controller.rb

```diff
@@ -31,6 +34,20 @@ def show
 
       private
 
+      def matching_adswords_urls
+        keywords_query = KeywordsQuery.new(Keyword, filter_params)
+        pagination, keywords = paginated_authorized(keywords_query.call)
+        keyword_presenters = keywords.map { |item| KeywordPresenter.new(item, filter_params) }
+
+        render json: KeywordSerializer.new(keyword_presenters, meta: meta_from_pagination(pagination))
```

Comment:
What do you think for queried Keywords we return a different Serializer than `all_keywords`. This way when we filter we have a different result than all?

Reply:
I did think about a different Serializer for all keywords and the queried ones. However, if a client wants some extra information like the name, id, or a full array of ads_top_urls (unfiltered), creating a new serializer will require the client to handle complex mapping.  If we return an extra `matching_adword_urls` client only need to create a new property in the model and mark it as optional.